Neo4j.Driver/Neo4j.Driver/Internal/Routing/RoundRobinLoadBalancer.cs

```diff
@@ -166,19 +166,22 @@ internal RoundRobinRoutingTable UpdateRoutingTable(Func<IPooledConnection, Round
                         return rediscoveryFunc == null ? Rediscovery(conn) : rediscoveryFunc.Invoke(conn);
                     }
                 }
-                catch (SessionExpiredException)
-                {
-                    // ignored
-                    // Already handled by connection pool error handler to remove from load balancer
-                }
-                catch (InvalidDiscoveryException)
+                catch (Exception e)
                 {
-                    // The result is invalid probably due to partition.
-                    _routingTable.Remove(uri);
+                    _logger?.Info($"Failed to update routing table with server uri={uri} due to error {e.Message}");
+                    if (e is SessionExpiredException)
```

Comment:
Can this branch be removed? It seems like we are only interested in `InvalidDiscoveryException` here.